ksqldb-engine/src/main/java/io/confluent/ksql/planner/plan/PullFilterNode.java

```diff
@@ -564,8 +613,21 @@ private Instant asInstant(final Expression other) {
         }
       }
 
+      final Object obj = new GenericExpressionResolver(
+          SqlTypes.BIGINT,
+          name,
+          metaStore,
+          ksqlConfig,
+          "pull query window bounds extractor",
+          ksqlConfig.getBoolean(KsqlConfig.KSQL_QUERY_PULL_INTERPRETER_ENABLED)
+      ).resolve(other);
+
+      if (obj instanceof Integer || obj instanceof Long) {
```

Comment:
Should we allow float/double to be casted to INT as well?

Reply:
The GenericExpressionResolver is in strict mode which I don't believe allows this.  We don't do a lot of conversions elsewhere automatically, such as with the key values (and would be a bit harder to generify), so I think this if fine for consistency.